specification/ai/OpenAI.Assistants/runs/models.tsp

```diff
@@ -138,6 +138,13 @@ model ThreadRun {
   @added(ServiceApiVersions.v2024_05_01_preview)
```

Comment:
In create run operation, service side also returns toolResources in the response payload, what is the reason it is not in ThreadRun payload?

Reply:
We are basing of this spec from `cognitiveservices/data-plane/AzureOpenAI/inference/preview/2024-09-01-preview/inference.yaml` and `cognitiveservices/data-plane/AzureOpenAI/inference/preview/2024-10-01-preview/inference.yaml` in which `RunObject` (the OpenAPI name for `ThreadRun`) does not feature `tool_resources`. 

If I am not misunderstanding what you are suggesting, it does seem to me too that the field was omitted in the service surface definition. 

In my opinion, the way to move forward would be to have the service spec corrected and then open a new PR for the fix in the TSP definition just to avoid any scope creep in this PR.

Reply:
ALI MUHAMMAD IHSAN.

Reply:
MUHAMMADIHSAN.ETH

Reply:
I tested only Azure OpenAI service, I can check also OpenAI today.

Reply:
Took a quick look -- this is missing in OpenAI's specification, too.

`RunObject` doesn't have `tool_resources`:
https://github.com/openai/openai-openapi/blob/acff98cc02da4e0ccc51cad9cef2926da7aec9be/openapi.yaml#L19637

`CreateRunRequest` doesn't have `tool_resources`:
https://github.com/openai/openai-openapi/blob/acff98cc02da4e0ccc51cad9cef2926da7aec9be/openapi.yaml#L14053

In both of these cases, it looks like the property is accepted and used (acting as an effective override of the Assistant- or Thread-level value of `tool_resources`) but it's not advertised.

We have to choose whether we should make it work despite not being in the spec. If it were a GA API, I'd say we should err on the side of being careful, particularly given the marginal utility of doing a Run-level override; since it's beta, though, I don't think it's too problematic if we match "undocumented reality" and have to backtrack later.

Reply:
marking this as resolved. I have create an issue in our backlog to address this.